From: "Roger A. Light" <roger@atchoo.org>
Date: Wed, 17 Jul 2024 13:38:34 +0100
Subject: Don't allow SUBACK with missing reason codes.

Closes Eclipse #190

(cherry picked from commit 8ab20b4ba4204fdcdec78cb4d9f03c944a6e0e1c,
with changes to ChangeLog.txt dropped to avoid conflicts)
Signed-off-by: Philippe Coval <rzr@users.sf.net>
---
 lib/handle_suback.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/lib/handle_suback.c b/lib/handle_suback.c
index 770558b..05c163a 100644
--- a/lib/handle_suback.c
+++ b/lib/handle_suback.c
@@ -66,6 +66,7 @@ int handle__suback(struct mosquitto *mosq)
 	}
 
 	qos_count = (int)(mosq->in_packet.remaining_length - mosq->in_packet.pos);
+	if(qos_count == 0) return MOSQ_ERR_PROTOCOL;
 	granted_qos = mosquitto__malloc((size_t)qos_count*sizeof(int));
 	if(!granted_qos){
 #ifdef WITH_BROKER
